src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.cpp

```diff
@@ -65,6 +65,48 @@ ONNXOpShapeHelper<OP>::ONNXOpShapeHelper(OP *newOp, int numResults,
   };
 }
 
+template <>
+Value ONNXOpShapeHelper<Operation>::getOutput(int n) {
+  return op->getResult(n);
+}
+
+template <class OP>
+DimsExpr &ONNXOpShapeHelper<OP>::dimsForOutput(int n) {
+  Value output = getOutput(n);
```

Comment:
@AlexandreEichenberger I found that this is good location to insert the combination code, in the sense that it minimizes the changes.